llvm/lib/Transforms/Utils/LKMMDependenceAnalysis.cpp

```diff
@@ -0,0 +1,2011 @@
+//===- LKMMDependenceAnalaysis.cpp - LKMM Deps Implementation -------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file implements two passes to determine whether data, addr and ctrl
+/// dependencies were preserved according to the Linux kernel memory model.
+///
+/// The first pass annotates relevant dependencies in unoptimized IR and the
+/// second pass verifies that the dependenices still hold in optimized IR.
+///
+/// Linux kernel memory model:
+/// https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/tools/memory-model/Documentation/explanation.txt
+///
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Transforms/Utils/LKMMDependenceAnalysis.h"
+#include "llvm/Analysis/CFG.h"
+#include "llvm/Analysis/PostDominators.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Metadata.h"
+#include "llvm/IR/Module.h"
+
+// TODO add clang support for clang -fsanitize=lkmm
+// TODO strict and relaxed mode
+
+namespace llvm {
+namespace {
+
+#define INTERPROCEDURAL_REC_LIMIT_ANNOTATION 3
```

Comment:
In C++ avoid constants via '#define'. 

https://abseil.io/tips/140

So probably 'constexpr int InterproceduralRecLimitAnnotation = 3;
...